File: AbyssalCreatures/src/main/java/com/offz/spigot/abyssalcreatures/Mobs/Passive/NPC.java
package com.offz.spigot.abyssalcreatures.Mobs.Passive;

import com.offz.spigot.mobzy.Builders.MobBuilder;
import com.offz.spigot.mobzy.Mobs.Types.PassiveMob;
import com.offz.spigot.mobzy.Pathfinders.PathfinderGoalLookAtPlayerPitchLock;
import net.minecraft.server.v1_13_R2.*;
import org.bukkit.Material;
import org.bukkit.entity.LivingEntity;

public class NPC extends PassiveMob {

    public NPC(World world, String name, int modelID) {
        super(world, new MobBuilder(name, modelID)
                .setModelMaterial(Material.DIAMOND_AXE));

        getBukkitEntity().setCustomName(name);
        setCustomNameVisible(true);
        setInvulnerable(true);
        setSize(0.6f, 0.6f);
        addScoreboardTag("npc");
        ((LivingEntity) this.getBukkitEntity()).setRemoveWhenFarAway(false);
    }

    //Stop from being pushed around
    @Override
    public void collide(Entity entity) {
    }

    @Override
    public void move(EnumMoveType enummovetype, double d0, double d1, double d2) {
    }

    @Override
    public void createPathfinders() {
        goalSelector.a(2, new PathfinderGoalRandomLookaround(this));
        goalSelector.a(7, new PathfinderGoalLookAtPlayerPitchLock(this, EntityTypes.PLAYER, 6.0F, 0.02F));
    }
}

File: AbyssalCreatures/src/main/java/com/offz/spigot/abyssalcreatures/mobs/passive/NPC.kt
package com.offz.spigot.abyssalcreatures.mobs.passive

import com.offz.spigot.mobzy.mobs.MobTemplate
import com.offz.spigot.mobzy.mobs.types.PassiveMob
import com.offz.spigot.mobzy.pathfinders.PathfinderGoalLookAtPlayerPitchLock
import net.minecraft.server.v1_15_R1.*
import org.bukkit.Material
import org.bukkit.entity.LivingEntity

class NPC(world: World?, name: String, modelID: Int) :
        PassiveMob(world, MobTemplate(name = name, modelID = modelID, modelMaterial = Material.DIAMOND_AXE)) {
    //Stop from being pushed around
    override fun move(enummovetype: EnumMoveType?, vec3d: Vec3D?) = Unit

    override fun collide(entity: Entity) = Unit

    //Prevent NPCs from getting damaged by anything
    override fun damageEntity(damagesource: DamageSource, f: Float) = false

    override fun createPathfinders() {
        addPathfinderGoal(2, PathfinderGoalRandomLookaround(this))
        addPathfinderGoal(7, PathfinderGoalLookAtPlayerPitchLock(this, EntityTypes.PLAYER, 6.0, 0.02f))
    }

    init {
        living.customName = name
        customNameVisible = true
        isInvulnerable = true
        addScoreboardTag("npc")
        (bukkitEntity as LivingEntity).removeWhenFarAway = false
    }
}
